### DISCUSSION TITLE
How to make 2 two motors run at the same time

### DISCUSSION THREAD

**Discussion by dani0716** [pybricks/support#1944] [Q&A]
Hello! I'm pretty new to pybricks but with the spike prime/ robot inventor hubs, is it possible to make multiple motors run at once instead of one at a time?

Thanks

**Comment by BertLindeman** [comment]
Sure, you can.

You can couple them in a drivebase if you want.
But separately they can be controlled as well.

Bert

**Reply by dani0716** [reply]
How would I control them separately so that multiple motors can run at once?

**Reply by BertLindeman** [reply]
Could you give us an idea what you want to achieve?
Maybe a short program what you tried?

We can now only guess what you want.

Are you using blocks or python is an important difference.

Some possibilities:
You can define more than one motor and in your program let them each do different tasks.
And you can use a set of motors as a drivebase.
Or define a car with motors.

Bert

**Reply by dani0716** [reply]
I'm trying to make a gantry robotic, an easier way to visualise it is like a 3d printer, like motors on the 3 different axis's move at the same time to improve efficiency, moving up, down and forwards one after the other is slow and will slow down the productivity rate. So I just wanna now how to control multiple motors at once. 

Could I just sure (wait=FALSE)?

**Reply by BertLindeman** [reply]
> I'm trying to make a gantry robotic, an easier way to visualise it is like a 3d printer, like motors on the 3 different axis's move at the same time to improve efficiency, moving up, down and forwards one after the other is slow and will slow down the productivity rate. So I just wanna now how to control multiple motors at once.
> 
> Could I just sure (wait=FALSE)?

Thank you. That makes a picture in my head.
I assume you will write that in python, because you mention the `wait=False`.

One syntax example from the doc:
```
run_angle(speed, rotation_angle, then=Stop.HOLD, wait=True)
```

Say you define 3 motors
- motor_x
- motor_y and
- motor_z

And some constants:
- hor_speed
- hor_angle_zero
- hor_angle_max
etc.

Give one motor a task e.g. horizontal from the angle to the max:
```python
motor_x.run_angle(hor_speed, hor_angle_max - motor_x.angle(), wait=False
```

And similar to that for the other motors.
After setting the motors, wait until all are ready and go on with the program.

Does that help?

Bert
